10up/classifai
# Feature/129 bulk process existing content

### Description of the Change



This PR adds the ability to Process Existing media. It relates to #129.

Notes:
- Due to the current design of `ComputerVision`, we have to combine alt text and tags generation into one `Scan Image` action.
- I used `filter_input` function instead of WordPress sanitization ones because ClassifAI is already using it in multiple places.

### Verification Process

1. Follow [this gist](https://gist.github.com/felipeelia/3821646676923a46c6eb68c65a4dd750) and [this comment](https://gist.github.com/felipeelia/3821646676923a46c6eb68c65a4dd750?permalink_comment_id=4229002#gistcomment-4229002) to proxy the local webserver.
2. Ensure the ClassifAI > Image Processing is set up with valid Computer Vision credentials.
3. Enable alt text, tags generation, and smart cropping.
4. Go to the media library, switch to list view.
5. See new bulk actions: `Scan Image` and `Smart Crop`.
6. Select some images > choose `Scan image` > Apply.
7. See the page reload, see the success notice containing the number of scanned images.
8. See the tag and alt text generated for selected images.
9. Select some images > choose `Smart crop` > Apply.
10. See the page reload, see the success notice containing the number of cropped images.
11. See the thumbnail cropped (You should test images with the main object on the side).
12. Disable Smart Crop > Don't see the `Smart Crop` bulk action.
13. Disable both alt text and tags generation > Don't see the `Scan Image` bulk actions.

### Changelog Entry



Added: scan and crop bulk actions for media files.
